src/linalg/spectral.jl

```diff
@@ -151,13 +151,22 @@ function incidence_matrix(g::AbstractGraph, T::DataType=Int; oriented=false)
     # every col has the same 2 entries
     colpt = collect(1:2:(nz + 1))
     nzval = repeat([(isdir || oriented) ? -one(T) : one(T), one(T)], n_e)
+    swap(i,j) = begin
+        tmp = nzval[i]
+        nzval[i] = nzval[j]
+        nzval[j] = tmp
+    end
```

Comment:
why can't we just do
```
nzval[i], nzval[j] = nzval[j], nzval[i]
```
and eliminate the need for `tmp`?

```
julia> a = 3
3

julia> b = 10
10

julia> a, b, = b, a
(10, 3)

julia> a
10

julia> b
3
```

Reply:
You do this below :)

Reply:
I did it with values, does that work with vectors? I'm thinking like a C programmer.

Reply:
```
julia> a = [1,2,3,4,5]
5-element Array{Int64,1}:
 1
 2
 3
 4
 5

julia> a[3], a[5] = a[5], a[3]
(5, 3)

julia> a
5-element Array{Int64,1}:
 1
 2
 5
 4
 3
```